src/pages/index.js

```diff
@@ -36,82 +82,213 @@ const editCardPopup = new PopupWithForm(
 );
+editAvatarPopup.setEventListeners();
 
 const popupWithImage = new PopupWithImage({
   popupSelector: "#previewModal",
 });
 popupWithImage.setEventListeners();
 
+const confirmPopup = new ConfirmPopup({
+  popupSelector: "#delete-modal",
+});
+confirmPopup.setEventListeners();
+
+// Instantiate form validators
 const editFormValidator = new FormValidator(settings, profileEditForm);
 const addFormValidator = new FormValidator(settings, addCardForm);
+editFormValidator.enableValidation();
+addFormValidator.enableValidation();
+
+// Helper Functions
+function createCard({ name, link, _id, isLiked }) {
+  // Instantiate the card
+  const card = new Card(
+    { name, link, _id, isLiked },
+    "#card-template",
+    handleImageClick,
+    handleLikeClick,
+    handleUnlikeClick,
+    handleDeleteCard
+  );
+
+  const cardElement = card.getView();
+  const cardContainer = document.querySelector(".modal__container");
+
+  cardContainer.appendChild(cardElement);
 
-// Render initial cards
-cardSection.renderItems();
+  const deleteButton = cardElement.querySelector(".card__delete-button");
 
-// Functions
+  if (deleteButton) {
+    deleteButton.addEventListener("click", () => {
+      confirmPopup.open();
+    });
+  } else {
+    console.error("Delete button not found within card template.");
+  }
 
-// Function to create a new card
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick); // Instantiate the new Card
-  const cardElement = card.getView(); // Get the card's HTML element
-  return cardElement; // Return the card's HTML element
+  return cardElement;
 }
 
-/* Event Handlers */
+//Temporary function to upload cards to server, if you want me to delete it
+// I can
 
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo({
-    name: formData.title,
-    description: formData.description,
+let cardUploaded = true;
+
+function uploadCardsToServer() {
+  if (cardUploaded) {
+    console.log("Cards already uploaded to server.");
+    return;
+  }
+
+  cards.forEach((card) => {
+    api
+      .createCard(card)
+      .then((addedCard) => {
+        console.log("Card added to server", addedCard);
+      })
+      .catch((err) => {
+        console.error("Error adding card to server", err);
+      });
   });
-  editCardPopup.close();
+}
+
+uploadCardsToServer();
+
+// Event Handlers
+
+function handleEditAvatarSubmit(formData) {
+  editAvatarPopup.renderLoading(true);
+  const avatarUrl = formData.avatarUrl;
+  api
+    .updateProPic(avatarUrl)
+    .then((updatedUserData) => {
+      console.log(updatedUserData);
+      userInfo.setuserAvatar(updatedUserData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error);
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoading(false, "Save");
+    });
+}
+
+function handleProfileEditSubmit(formData) {
+  editCardPopup.renderLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```